WooCommerce/Classes/ViewRelated/Orders/Collect Payments/CollectOrderPaymentUseCase.swift

```diff
@@ -494,10 +503,34 @@ private extension CollectOrderPaymentUseCase {
             onCompleted()
         }
     }
+    /// Allow merchants to print or email backend-generated receipts.
+    /// The alerts presenter can be simplified once we remove legacy receipts: https://github.com/woocommerce/woocommerce-ios/issues/11897
+    ///
+    func presentBackendReceiptAlert(alertProvider paymentAlerts: CardReaderTransactionAlertsProviding, onCompleted: @escaping () -> ()) {
+        // Present receipt alert
+        alertsPresenter.present(viewModel: paymentAlerts.success(printReceipt: { [weak self] in
+            guard let self = self else { return }
 
-    /// Allow merchants to print or email the payment receipt.
+            self.paymentOrchestrator.presentBackendReceipt(for: self.order, onCompletion: { receipt in
+                self.presentBackendReceiptModally(receipt: receipt, onCompleted: onCompleted)
```

Comment:
This is strongly capturing `self`... is that what we need/want?

Reply:
It's not! I was considering if breaking the reference cycle on the outer closure would be enough, since this inner `self` would be strongly captured, but encapsulated, on the lifetime of the outer closure. I don't see a reason for not  making it weak thought, is just safer. Fixed on https://github.com/woocommerce/woocommerce-ios/pull/11896/commits/2990cf5a9c836e05cfb25817b45e6bc0904bad55

Reply:
> I was considering if breaking the reference cycle on the outer closure would be enough, since this inner self would be strongly captured, but encapsulated, on the lifetime of the outer closure

You're right, and in practice, it probably would be fine because of that. With the previous code, `self` should only be captured when the inner code runs.